ouroboros-network/src/Ouroboros/Network/Socket.hs

```diff
@@ -129,6 +129,17 @@ hexDump :: BL.ByteString -> String -> IO ()
 hexDump buf out | BL.empty == buf = say out
 hexDump buf out = hexDump (BL.tail buf) (out ++ printf "0x%02x " (BL.head buf))
 
+-- | The enumeration of all mini-protocols in our demo protocol.
+data DemoProtocols = ChainSync
+  deriving (Eq, Ord, Enum, Bounded)
+
+instance Mx.ProtocolEnum DemoProtocols where
+  fromProtocolEnum ChainSync = 0
```

Comment:
As in Pipe.hs, this must be 2. 0 is MuxControl.

Reply:
Yep, thanks!